src/treelearner/gpu_tree_learner.cpp

```diff
@@ -0,0 +1,1064 @@
+    if (!is_feature_used[feature_index]) { continue; }
+    const int tid = omp_get_thread_num();
+    SplitInfo smaller_split;
+#if GPU_DEBUG >= 3
+    size_t bin_size = train_data_->FeatureNumBin(feature_index) + 1; 
+    printf("feature %d smaller leaf (before fix):\n", feature_index);
+    PrintHistograms(smaller_leaf_histogram_array_[feature_index].RawData() - 1, bin_size);
+#endif
+
+    train_data_->FixHistogram(feature_index,
+                              smaller_leaf_splits_->sum_gradients(), smaller_leaf_splits_->sum_hessians(),
+                              smaller_leaf_splits_->num_data_in_leaf(),
+                              smaller_leaf_histogram_array_[feature_index].RawData());
+
+#if GPU_DEBUG >= 3
+    printf("feature %d smaller leaf:\n", feature_index);
+    PrintHistograms(smaller_leaf_histogram_array_[feature_index].RawData() - 1, bin_size);
+#endif
+    smaller_leaf_histogram_array_[feature_index].FindBestThreshold(
+      smaller_leaf_splits_->sum_gradients(),
+      smaller_leaf_splits_->sum_hessians(),
+      smaller_leaf_splits_->num_data_in_leaf(),
+      &smaller_split);
+    if (smaller_split.gain > smaller_best[tid].gain) {
+      smaller_best[tid] = smaller_split;
+      smaller_best[tid].feature = train_data_->RealFeatureIndex(feature_index);
+    }
+    // only has root leaf
+    if (larger_leaf_splits_ == nullptr || larger_leaf_splits_->LeafIndex() < 0) { continue; }
+
+    if (use_subtract) {
+      larger_leaf_histogram_array_[feature_index].Subtract(smaller_leaf_histogram_array_[feature_index]);
+    } else {
+      train_data_->FixHistogram(feature_index, larger_leaf_splits_->sum_gradients(), larger_leaf_splits_->sum_hessians(),
+                                larger_leaf_splits_->num_data_in_leaf(),
+                                larger_leaf_histogram_array_[feature_index].RawData());
+    }
+#if GPU_DEBUG >= 4
+    printf("feature %d larger leaf:\n", feature_index);
+    PrintHistograms(larger_leaf_histogram_array_[feature_index].RawData() - 1, bin_size);
+#endif
+    SplitInfo larger_split;
+    // find best threshold for larger child
+    larger_leaf_histogram_array_[feature_index].FindBestThreshold(
+      larger_leaf_splits_->sum_gradients(),
+      larger_leaf_splits_->sum_hessians(),
+      larger_leaf_splits_->num_data_in_leaf(),
+      &larger_split);
+    if (larger_split.gain > larger_best[tid].gain) {
+      larger_best[tid] = larger_split;
+      larger_best[tid].feature = train_data_->RealFeatureIndex(feature_index);
+    }
+  }
+
+  auto smaller_best_idx = ArrayArgs<SplitInfo>::ArgMax(smaller_best);
+  int leaf = smaller_leaf_splits_->LeafIndex();
+  best_split_per_leaf_[leaf] = smaller_best[smaller_best_idx];
+
+  if (larger_leaf_splits_ != nullptr && larger_leaf_splits_->LeafIndex() >= 0) {
+    leaf = larger_leaf_splits_->LeafIndex();
+    auto larger_best_idx = ArrayArgs<SplitInfo>::ArgMax(larger_best);
+    best_split_per_leaf_[leaf] = larger_best[larger_best_idx];
+  }
```

Comment:
If yes, can you just call SerialTreelearner::FindBestThresholds(), and write the additional codes for debugging?